query-planner-js/src/composedSchema/removeInaccessibleElements.ts

```diff
@@ -38,6 +39,27 @@ export function removeInaccessibleElements(
     }),
   );
 
+  // remove unions which have no accessible types
+  // this is a loop since unions can contain other unions
+  let shouldRevisitSchema = true;
```

Comment:
`shouldRevisitSchema` seems a bit generic as a variable name, because it's only meant to affect this particular loop. Perhaps something like `hasInaccessibleUnionMembers`, and then changing this to a `do ... while` loop?

Alternatively, perhaps you could extract all of this into a separate function, like `removeInaccessibleUnionMembers`?